flow-plugins/flow-maven-plugin/src/main/java/com/vaadin/flow/plugin/maven/FlowModeAbstractMojo.java

```diff
@@ -253,6 +253,38 @@ public static List<String> getClasspathElements(MavenProject project) {
         }
     }
 
+    /**
+     * Checks if Hilla is available based on the Maven project's classpath.
+     *
+     * @param mavenProject
+     *            Target Maven project
+     * @return true if Hilla is available, false otherwise
+     */
+    public static boolean isHillaAvailable(MavenProject mavenProject) {
+        List<String> classpathElements = FlowModeAbstractMojo
+                .getClasspathElements(mavenProject);
+        var resource = BuildFrontendUtil.getClassFinder(classpathElements)
+                .getResource("com/vaadin/hilla/EndpointController.class");
+        return resource != null;
```

Comment:
```suggestion
        ClassFinder classFinder = BuildFrontendUtil.getClassFinder(classpathElements);
        return classFinder.getResource("com/vaadin/hilla/EndpointController.class") != null;
```
Could be easier to read to take the classFinder and then check the resource.